internal/dependency/registry/local.go

```diff
@@ -0,0 +1,75 @@
+package registry
+
+import (
+	"github.com/g2a-com/klio/internal/dependency"
+	"github.com/g2a-com/klio/internal/log"
+	"github.com/spf13/afero"
+	"gopkg.in/yaml.v3"
+	"io/ioutil"
+	"strings"
+)
+
+// local represents registry hosted locally
+type local struct {
+	path  string
+	index Index
+	fs    afero.Fs
+}
+
+// NewLocal returns new registry instance hosted locally
+func NewLocal(registryPath string) Registry {
+	registry := &local{
+		path: registryPath,
+		fs:   afero.NewOsFs(),
+	}
+
+	return registry
+}
+
+func (reg *local) Update() error {
+	log.Spamf("Loading registry: %s", reg.path)
+
+	var buffer []byte
+
+	path := strings.TrimPrefix(reg.path, "file://")
+	file, err := reg.fs.Open(path)
+	if err != nil {
+		return err
+	}
+	buffer, err = ioutil.ReadAll(file)
```

Comment:
`ioutil` is deprecated since 1.16 (or at least "not preferred")  - use `io.ReadAll`, or `afero.ReadFile`:

```go
buffer, err = afero.ReadFile(reg.fs, path)
```